src/ripple/net/impl/HTTPDownloader.cpp

```diff
@@ -286,16 +275,26 @@ HTTPDownloader::do_session(
     JLOG(j_.trace()) << "download completed: " << dstPath.string();
 
     close(p);
-    exit();
+    exit(true);
+}
+
+void
+HTTPDownloader::onStop()
+{
+    std::unique_lock lock(m_);
 
-    // Notify the completion handler
-    complete(std::move(dstPath));
+    cancelDownloads_ = true;
```

Comment:
Can this assignment occur before the lock? Same with evaluation in ```download```.

Reply:
Yes, done